src/statsdc.c

```diff
@@ -44,7 +45,8 @@ _statsdc_error(statsdc_t sdc, const char *format, ...) {
 	}
 
 	va_start(ap, format);
-	vasprintf(&(sdc->last_error_string), format, ap);
+	if (-1 == vasprintf(&(sdc->last_error_string), format, ap))
+		errx(1, "cannot allocate memory for error message");
```

Comment:
`vasprintf` could fail for other reasons; would it be better to make the error message less specific?
(Same for the other `*as*print*` calls.)

Reply:
The GNU documentation does not detail other failures that could occur and the man page for FreeBSD only mentions `malloc(3)` as a failure mode, which is why I went with this message.

Reply:
The documentation I have lists "others" as a possibility:

> If memory allocation wasn't possible, or some other error occurs,
> these functions will return -1, and the contents of strp is
> undefined.

🤷‍♀

Reply:
I would like to remove the `last_error_string` field to get rid of all this headache. (I do not understand why we need `last_error_string` and why it should be customizable.)

But 👍 for replacing autoconf with a simple makefile.

Reply:
@pjhades let's remove the `last_error_string` field in another PR.

Reply:
@gnuvince Perhaps something like `errno` would be enough. Then we could have a table of pre-defined error messages. 😮